### ISSUE TITLE
Error in Go routine: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by Sarke** [containous/traefik#5828]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Tried to add a load-balancer service for A-B testing.

### What did you expect to see?

No panic

### What did you see instead?

Panic

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.0.4
Codename:     montdor
Go version:   go1.13.3
Built:        2019-10-28T20:23:57Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Docker Swarm.

```yaml
environment:
  TRAEFIK_ENTRYPOINTS_WEB_ADDRESS: ":80"
  TRAEFIK_ENTRYPOINTS_WEB_FORWARDEDHEADERS_TRUSTEDIPS: "173.245.48.0/20,103.21.244.0/22,103.22.200.0/22,103.31.4.0/22,141.101.64.0/18,108.162.192.0/18,190.93.240.0/20,188.114.96.0/20,197.234.240.0/22,198.41.128.0/17,162.158.0.0/15,104.16.0.0/12,172.64.0.0/13,131.0.72.0/22,2400:cb00::/32,2606:4700::/32,2803:f800::/32,2405:b500::/32,2405:8100::/32,2a06:98c0::/29,2c0f:f248::/32"
  TRAEFIK_ENTRYPOINTS_WEBSECURE_ADDRESS: ":443"
  TRAEFIK_ENTRYPOINTS_WEBSECURE_FORWARDEDHEADERS_TRUSTEDIPS: "173.245.48.0/20,103.21.244.0/22,103.22.200.0/22,103.31.4.0/22,141.101.64.0/18,108.162.192.0/18,190.93.240.0/20,188.114.96.0/20,197.234.240.0/22,198.41.128.0/17,162.158.0.0/15,104.16.0.0/12,172.64.0.0/13,131.0.72.0/22,2400:cb00::/32,2606:4700::/32,2803:f800::/32,2405:b500::/32,2405:8100::/32,2a06:98c0::/29,2c0f:f248::/32"
  TRAEFIK_ENTRYPOINTS_TRAEFIK: "false"
  TRAEFIK_API: "true"
  TRAEFIK_LOG: "true"
  TRAEFIK_LOG_LEVEL: INFO
  TRAEFIK_LOG_FORMAT: "json"
  TRAEFIK_ACCESSLOG: "true"
  TRAEFIK_ACCESSLOG_FORMAT: "json"
  TRAEFIK_ACCESSLOG_FILEPATH: "/root/logs/traefik/access.log"
  TRAEFIK_ACCESSLOG_BUFFERINGSIZE: 10
  TRAEFIK_PING: "true"
  TRAEFIK_PROVIDERS_DOCKER: "true"
  TRAEFIK_PROVIDERS_DOCKER_NETWORK: proxy
  TRAEFIK_PROVIDERS_DOCKER_SWARMMODE: "true"
  TRAEFIK_PROVIDERS_DOCKER_EXPOSEDBYDEFAULT: "false"
  TRAEFIK_PROVIDERS_FILE_DIRECTORY: "/etc/traefik/conf.d/"
  TRAEFIK_METRICS_STATSD: "true"
  TRAEFIK_METRICS_STATSD_ADDRESS: "influxdb_telegraf:8125"
```

This seems to be the offending TOML.  When I add this in a separate file it causes the error.

```toml
[http.services.php73-varnish-split-production]
  [[http.services.php73-varnish-split-production.weighted.services]]
    name = "php73-production@docker"
    weight = 10
  [[http.services.php73-varnish-split-production.weighted.services]]
    name = "varnish@docker"
    weight = 10
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

End of the debug log below.

```json
{
  "entryPointName": "websecure",
  "level": "debug",
  "middlewareName": "tracing",
  "middlewareType": "TracingForwarder",
  "msg": "Added outgoing tracing middleware php73",
  "routerName": "php73@docker",
  "time": "2019-11-14T05:58:42Z"
}
{
  "entryPointName": "websecure",
  "level": "debug",
  "middlewareName": "traefik-internal-recovery",
  "middlewareType": "Recovery",
  "msg": "Creating middleware",
  "time": "2019-11-14T05:58:42Z"
}
{
  "entryPointName": "websecure",
  "level": "debug",
  "middlewareName": "metrics-entrypoint",
  "middlewareType": "Metrics",
  "msg": "Creating middleware",
  "time": "2019-11-14T05:58:42Z"
}
{
  "entryPointName": "websecure",
  "level": "debug",
  "middlewareName": "metrics-entrypoint",
  "middlewareType": "Metrics",
  "msg": "Creating middleware",
  "time": "2019-11-14T05:58:42Z"
}
{
  "entryPointName": "traefik",
  "level": "debug",
  "middlewareName": "metrics-entrypoint",
  "middlewareType": "Metrics",
  "msg": "Creating middleware",
  "time": "2019-11-14T05:58:42Z"
}
{
  "entryPointName": "web",
  "level": "debug",
  "middlewareName": "metrics-entrypoint",
  "middlewareType": "Metrics",
  "msg": "Creating middleware",
  "time": "2019-11-14T05:58:42Z"
}
{
  "level": "error",
  "msg": "Error in Go routine: runtime error: invalid memory address or nil pointer dereference",
  "time": "2019-11-14T05:58:42Z"
}
{
  "level": "error",
  "msg": "
    Stack: goroutine 73 [running]:
    runtime/debug.Stack(0xc000112000, 0x24d0e85, 0x17)
      /usr/local/go/src/runtime/debug/stack.go:24 +0x9d
    github.com/containous/traefik/v2/pkg/safe.defaultRecoverGoroutine(0x209b8c0, 0x3ee5610)
      /go/src/github.com/containous/traefik/pkg/safe/routine.go:160 +0xb1
    github.com/containous/traefik/v2/pkg/safe.GoWithRecover.func1.1(0x25ab3a0)
      /go/src/github.com/containous/traefik/pkg/safe/routine.go:150 +0x57
    panic(0x209b8c0, 0x3ee5610)
      /usr/local/go/src/runtime/panic.go:679 +0x1b2
    github.com/containous/traefik/v2/pkg/metrics.OnConfigurationUpdate(0xc0007bc270, 0xc0003e0e80, 0x3, 0x4)
      /go/src/github.com/containous/traefik/pkg/metrics/prometheus.go:232 +0x618
    github.com/containous/traefik/v2/pkg/server.(*Server).loadConfiguration(0xc0001f6000, 0x24aa889, 0x4, 0xc000995840)
      /go/src/github.com/containous/traefik/pkg/server/server_configuration.go:59 +0x49a
    github.com/containous/traefik/v2/pkg/server.(*Server).listenConfigurations(0xc0001f6000, 0xc00008c230)
      /go/src/github.com/containous/traefik/pkg/server/server_configuration.go:249 +0x51
    github.com/containous/traefik/v2/pkg/server.(*Server).Start.func3(0xc00008c230)
      /go/src/github.com/containous/traefik/pkg/server/server.go:183 +0x34
    github.com/containous/traefik/v2/pkg/safe.(*Pool).Go.func1()
      /go/src/github.com/containous/traefik/pkg/safe/routine.go:90 +0x7b
    github.com/containous/traefik/v2/pkg/safe.GoWithRecover.func1(0x25ab3a0, 0xc00047dc80)
      /go/src/github.com/containous/traefik/pkg/safe/routine.go:153 +0x57
    created by github.com/containous/traefik/v2/pkg/safe.GoWithRecover
      /go/src/github.com/containous/traefik/pkg/safe/routine.go:147 +0x49
  ",
  "time": "2019-11-14T05:58:42Z"
}
```

**Comment by Sarke** [comment]
Offending line:

https://github.com/containous/traefik/blob/v2.0.4/pkg/metrics/prometheus.go#L232

Of note is that we use StatsD and InfluxDB, but not Prometheus.

**Comment by ldez** [comment]
Duplicate of #5755
Fixed by #5759
